BlueprintUICommonControls/Sources/EditingMenu.swift

```diff
@@ -0,0 +1,309 @@
+    ///     }
+    ///
+    ///     EditingMenuItem(.select) {
+    ///         print("Selected!")
+    ///     }
+    /// }
+    /// ```
+    public func editingMenu(
+        @Builder<EditingMenuItem> with items: () -> [EditingMenuItem]
+    ) -> EditingMenu {
+        EditingMenu(wrapping: self, with: items)
+    }
+}
+
+extension EditingMenuItem.Kind {
+
+    fileprivate var selector: Selector {
+
+        let actions = UIResponderStandardEditActions.self
+
+        switch self {
+        case .cut:
+            return #selector(actions.cut)
+        case .copy:
+            return #selector(actions.copy)
+        case .paste:
+            return #selector(actions.paste)
+        case .delete:
+            return #selector(actions.delete)
+        case .select:
+            return #selector(actions.select)
+        case .selectAll:
+            return #selector(actions.selectAll)
+        case .toggleBoldface:
+            return #selector(actions.toggleBoldface)
+        case .toggleItalics:
+            return #selector(actions.toggleItalics)
+        case .toggleUnderline:
+            return #selector(actions.toggleUnderline)
+        case .increaseSize:
+            return #selector(actions.increaseSize)
+        case .decreaseSize:
+            return #selector(actions.decreaseSize)
+
+        case .printContent:
+            guard #available(iOS 15.0, *) else { fatalError() }
+            return #selector(actions.printContent)
+        case .pasteAndGo:
+            guard #available(iOS 15.0, *) else { fatalError() }
+            return #selector(actions.pasteAndGo)
+        case .pasteAndMatchStyle:
+            guard #available(iOS 15.0, *) else { fatalError() }
+            return #selector(actions.pasteAndMatchStyle)
+        case .pasteAndSearch:
+            guard #available(iOS 15.0, *) else { fatalError() }
+            return #selector(actions.pasteAndSearch)
+        }
+    }
+}
+
+
+extension EditingMenu {
+
+    fileprivate final class View: UIView {
+
+        var items: [EditingMenuItem] = []
+
+        private let longPress: UILongPressGestureRecognizer
+
+        override init(frame: CGRect) {
+            longPress = UILongPressGestureRecognizer()
+
+            super.init(frame: frame)
+
+            longPress.addTarget(self, action: #selector(handleLongPress))
+
+            addGestureRecognizer(longPress)
+        }
+
+        @available(*, unavailable)
+        required init?(coder: NSCoder) { fatalError() }
+
+        // MARK: UIView
+
+        override var canBecomeFirstResponder: Bool {
+            true
+        }
+
+        @objc override func canPerformAction(_ action: Selector, withSender sender: Any?) -> Bool {
+            items.map(\.kind.selector).contains(action)
+        }
+
+        // MARK: UIResponderStandardEditActions
+
+        @objc override func cut(_ any: Any?) {
+            perform(kind: .cut)
+        }
+
+        @objc override func copy(_ any: Any?) {
+            perform(kind: .copy)
```

Comment:
Do you think we should offer any default implementations? If not, I think the demo app should at illustrate actually copying, and "pasting" to the console

Reply:
Ah good idea! I can do that.